serverlessworkflow/modules/ROOT/pages/cloud/operator/referencing-resource-files.adoc

```diff
@@ -0,0 +1,155 @@
+:description: Referencing external resource files in a workflow definition
+:keywords: workflow, serverless, operator, kubernetes, minikube, configmap
+// links
+:kubernetes_create_configmap_url: https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#create-a-configmap
+:kubernetes_configmap_url: https://kubernetes.io/docs/concepts/configuration/configmap/
+
+This document describes how to reference resources files in the `SonataFlow` Custom Resource (CR).
+
+Most of the time, a workflow definition will required not only the flow definition, but also specification descriptors, schemas, subflows definitions, and etc.
+For example, when doing xref:service-orchestration/orchestration-of-openapi-based-services.adoc[service orchestration using OpenAPI descriptors], you need to tell the workflow where to find these descriptors in your context.
+
+If these resources are not in a remote location that can be accessed via the HTTP protocol, you must describe in the `SonataFlow` CR where to find them within the cluster. This is done via link:{kubernetes_configmap_url}[`ConfigMaps`].
+
+== Creating ConfigMaps with Workflow resources
+
+.Prerequisites
+* You have the resources available in your file system
+* You have permissions to create `ConfigMaps` in the target namespace
+
+Given that you already have the resource you want to add to your workflow definition, you link:{kubernetes_create_configmap_url}[can create a `ConfigMap`] as you normally would with the contents of the file.
+
+For example, given the following workflow:
+
+.Example of a workflow referencing external resources
+[source,yaml,subs="attributes+"]
+----
+apiVersion: sonataflow.org/v1alpha08
+kind: SonataFlow
+metadata:
+  name: service
+  annotations:
+    sonataflow.org/description: Hello Service!
+    sonataflow.org/version: 0.0.1
+    sonataflow.org/profile: dev
+spec:
+  flow:
+    start: Service
+    dataInputSchema: specs/workflow-service-schema.json <1>
+    functions:
+    - name: isWinner
+      operation: specs/workflow-service-openapi.json#isWinner <2>
+      type: rest
+    states:
+    - name: Service
+      type: operation
+      actions:
+      - name: CallService
+        functionRef:
+          refName: isWinner
+    end: true
+----
+
+<1> The workflow defines an input schema
+<2> The workflow requires an OpenAPI specification file to make a REST invocation
+
+For this example, you have two options. You can either create two `ConfigMaps` to have a clear separation of concerns or only one with both files.
+
+From the operator perspective, it won't make any difference since both files will be available for the workflow application in runtime.
+
+To make it simple, you can create only one `ConfigMap`. Given that the files are available in the current directory:
+
+.Creating a ConfigMap from the current directory
+[source,bash,subs="attributes+"]
+----
+kubectl create configmap service-resources --from-file=$(pwd) -n <workflow-namespace>
```

Comment:
$(pwd) might not be a good idea.
if user are running this command from a dir that have more files, that can lead the user to have issues.
maybe enforce to create a directory and add the files there would be nice.

Reply:
But that's the point. They created a new dir in the past command, added the files in the current dir and create the ConfigMap.